tests/TableViewTest.cc

```diff
@@ -0,0 +1,199 @@
+
+    auto count = 20;
+    for (int i = 0; i < count; ++i) {
+        auto msg = MessageBuilder()
+                       .setPartitionKey("key" + std::to_string(i))
+                       .setContent("value" + std::to_string(i))
+                       .build();
+        ASSERT_EQ(ResultOk, producer.send(msg));
+    }
+
+    // Create table view failed, The schema is not compatible
+    TableViewConfiguration tableViewConfiguration;
+    TableView tableView;
+    ASSERT_EQ(ResultOk, client.createTableView(topic, tableViewConfiguration, tableView));
+    ASSERT_EQ(tableView.size(), count);
+
+    // Set the v of k1 is empty
+    auto msg = MessageBuilder().setPartitionKey("key1").setContent("").build();
+    ASSERT_EQ(ResultOk, producer.send(msg));
+    waitUntil(
+        std::chrono::seconds(2), [&] { return tableView.size() == count - 1; }, 1000);
+
+    // assert interfaces.
+    std::string value;
+    ASSERT_TRUE(!tableView.containsKey("key1"));
+    ASSERT_TRUE(!tableView.getValue("key1", value));
+    ASSERT_TRUE(value.empty());
+
+    client.close();
+}
+
+TEST(TableViewTest, testNotSupportNonPersistentTopic) {
+    const std::string topic = TopicDomain::NonPersistent +
+                              "://public/default/testNotSupportNonPersistentTopic" +
+                              std::to_string(time(nullptr));
+    Client client(lookupUrl);
+
+    // Create table view failed, The schema is not compatible
+    TableViewConfiguration tableViewConfiguration;
+    TableView tableView;
+    ASSERT_EQ(ResultNotAllowedError, client.createTableView(topic, tableViewConfiguration, tableView));
+    client.close();
+}
+
+TEST(TableViewTest, testMultiTopicAndAutoUpdatePartitions) {
+    std::string uniqueTimeStr = std::to_string(time(nullptr));
+    std::string topic = "persistent://public/default/testMultiTopicAndAutoUpdatePartitions" + uniqueTimeStr;
+    Client client(lookupUrl);
+
+    std::string url = adminUrl + "admin/v2/persistent/public/default/testMultiTopicAndAutoUpdatePartitions" +
+                      uniqueTimeStr + "/partitions";
+    int res = makePutRequest(url, "5");
+    LOG_INFO("res = " << res);
+    ASSERT_FALSE(res != 204 && res != 409);
+
+    ProducerConfiguration producerConfiguration;
+    Producer producer;
+    ASSERT_EQ(ResultOk, client.createProducer(topic, producerConfiguration, producer));
+
+    auto count = 20;
+    for (int i = 0; i < count; ++i) {
+        auto msg = MessageBuilder()
+                       .setPartitionKey("key" + std::to_string(i))
+                       .setContent("value" + std::to_string(i))
+                       .build();
+        ASSERT_EQ(ResultOk, producer.send(msg));
+    }
+
+    TableViewConfiguration tableViewConfiguration;
+    TableView tableView;
+    // TODO need support multiReader first.
+    ASSERT_EQ(ResultOperationNotSupported, client.createTableView(topic, tableViewConfiguration, tableView));
```

Comment:
We can create multi-reader in the table view. Each partition owns one reader.

Reply:
I think we can support `MultiTopicsReader` directly(I'm doing it). In this way, the table view side does not need to make any changes.

Reply:
Support by https://github.com/apache/pulsar-client-cpp/pull/154